src/txParsers.ts

```diff
@@ -256,34 +278,73 @@ export const parseTxBody = (unparsedTxBody: unknown): TransactionBody => {
         metadataHash: parseOptional(unparsedTxBody.get(7), parseMetadataHash),
         validityIntervalStart: parseOptional(unparsedTxBody.get(8), parseValidityIntervalStart),
         mint: parseOptional(unparsedTxBody.get(9), parseMint),
+        scriptDataHash: parseOptional(unparsedTxBody.get(11), parseScriptDataHash),
+        collaterals: parseOptional(unparsedTxBody.get(13), parseCollaterals),
+        requiredSigners: parseOptional(unparsedTxBody.get(14), parseRequiredSigners),
+        networkId: parseOptional(unparsedTxBody.get(15), parseNetworkId),
     }
 }
 
 export const parseTx = (unparsedTx: unknown): Transaction => {
-    const [body, witnessSet, auxiliaryData] = parseTuple(
+    const [body, ...otherItems] = parseTuple(
         unparsedTx,
         ParseErrorReason.INVALID_TX_CBOR,
         parseTxBody,
         dontParse,
         dontParse,
+        dontParse,
     )
+    const presentItems = otherItems.filter((item) => item !== undefined)
 
-    return {body, witnessSet, auxiliaryData}
+    // cardano-cli with --shelley-era, --allegra-era and --mary-era
+    // includes only txBody, witnessSet and auxiliaryData
+    if (presentItems.length === 2) {
+        return { body, witnessSet: presentItems[0], auxiliaryData: presentItems[1] }
+    }
+
+    // cardano-cli with --alonzo-era includes all tx items
+    return {
+        body,
+        witnessSet: presentItems[0],
+        scriptValidity: presentItems[1],
+        auxiliaryData: presentItems[2],
+    }
 }
 
 export const parseRawTx = (unparsedRawTx: unknown): RawTransaction => {
-    const [body, item1, item2] = parseTuple(
+    const [body, ...otherItems] = parseTuple(
         unparsedRawTx,
         ParseErrorReason.INVALID_RAW_TX_CBOR,
         parseTxBody,
         dontParse,
         dontParse,
+        dontParse,
+        dontParse,
+        dontParse,
     )
-    // older versions of cardano-cli did not include scriptWitnesses
-    if (item2 === undefined) {
-        return {body, scriptWitnesses: undefined, auxiliaryData: item1}
+    const presentItems = otherItems.filter((item) => item !== undefined)
+
+    // older versions of cardano-cli included only txBody and auxiliaryData
+    if (presentItems.length === 1) {
+        return { body, auxiliaryData: presentItems[0] }
     }
+
     // cardano-cli expects indefinite-length scriptWitnesses
-    (item1 as any).encodeCBOR = Encoder.encodeIndefinite
-    return {body, scriptWitnesses: item1, auxiliaryData: item2}
+    (presentItems[0] as any).encodeCBOR = Encoder.encodeIndefinite
```

Comment:
Isn't this needed for other `presentItems`?

Reply:
No, another surprise egg from cardano-cli :) Example txBody:
```
[
    {
        0: [...txInputs...],
        13: [],
        1: [...txOutputs...],
        2: 1000000_2,
        14: [],
    },
    [_ ],
    [],
    [],
    true,
    null,
]
```